src/main.ino

```diff
@@ -348,20 +337,26 @@ void do_trial( size_t trial_num, int cs_type, bool isprobe = false )
     print_trial_info( );
     trial_start_time_ = millis( );
 
-    /*-----------------------------------------------------------------------------
-     *  PRE. Start imaging;  for 8 seconds.
-     *-----------------------------------------------------------------------------*/
-    unsigned duration = 8000;
+   /*-----------------------------------------------------------------------------
+    * PRE. Start imaging; 
+    * Usually durations of PRE_ is 8000 ms. For some trials is it randomly
+    * chosen between 6000 ms and 8000 ms. See protocol.h file and
+    * ./Protocols/BehaviourProtocols.xlsx file.
+    *-----------------------------------------------------------------------------*/
+    size_t duration = random(PROTO_PreStimDuration_LOW, PROTO_PreStimDuration_HIGH+1);
+
+    // From Ananth. He suggested that 60ms delay is required for every protocol. This is 
+    // apprently shutter delay for the camera and only required for trial number
+    // 1.
     if (trial_num == 1)
```

Comment:
@soumyaaymous For first trial there is additional delay of 60ms. This is suggested by @ananthamurthy . See the note in source.

Reply:
Alright.